keyboards/owlab/spring/spring.c

```diff
@@ -0,0 +1,127 @@
+/* 
+Copyright 2021 OwLab
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "spring.h"
+#include "rgblight.h"
+
+enum caps_modes{
+    CAPS_MODE_UPPER = 0, //UPPER CASE 
+    CAPS_MODE_LOWER      //LOWER CASE
+};
+
+uint8_t caps_mode_index;
+uint8_t pre_hue, pre_sat, pre_val; 
+uint8_t previous_rgb_mode = 0;  
```

Comment:
It may be simpler to just do:
```suggestion
rgblight_config_t prev_config;
```